sunpy/physics/differential_rotation.py

```diff
@@ -142,3 +147,163 @@ def solar_rotate_coordinate(coordinate, new_observer_time, new_observer_location
+    This is an inverse function needed by the scikit-image `transform.warp`
+    function.
+
+    Parameters
+    ----------
+    xy :
+        Array from `transform.warp`
+    smap : `~sunpy.map`
+        Original map that we want to transform
+    dt : `~astropy.units.Quantity`
+        Desired interval to rotate the input map by solar differential rotation.
+
+    Returns
+    -------
+    xy2 : `~numpy.ndarray`
+        Array with the inverse transformation
+    """
+    rotated_time = smap.date - timedelta(seconds=dt.to(u.s).value)
+
+    # Calculate the hpc coords
+    x = np.arange(0, smap.dimensions.x.value)
+    y = np.arange(0, smap.dimensions.y.value)
+    xx, yy = np.meshgrid(x, y)
+    hpc_coords = smap.pixel_to_world(xx * u.pix, yy * u.pix)
+
+    rotated_coord = solar_rotate_coordinate(hpc_coords, rotated_time)
+
+    # To find the values that are behind the sun we need to convert them to HeliographicStonyhurst
+    findOccult = rotated_coord.transform_to(HeliographicStonyhurst)
+    # and find which ones are outside the [-90, 90] range.
+    occult = np.logical_or(np.less(findOccult.lon, -90 * u.deg), np.greater(findOccult.lon, 90 * u.deg))
+
+    # NaN-ing values on the other side of the sun
+    rotated_coord.data.lon[occult] = np.nan * u.deg
+    rotated_coord.data.lat[occult] = np.nan * u.deg
+    rotated_coord.cache.clear()
+
+    # Go back to pixel co-ordinates
+    x2, y2 = smap.world_to_pixel(rotated_coord)
+
+    # Re-stack the data to make it correct output form
+    xy2 = np.column_stack([x2.value.flat, y2.value.flat])
+
+    # Returned a masked array with the non-finite entries masked.
+    return np.ma.array(xy2, mask=np.isinf(xy2))
+
+
+@u.quantity_input(dt=u.s)
+def diffrot_map(smap, dt):
+    """
+    Function to apply solar differential rotation to a sunpy map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map`
+        Original map that we want to transform.
+    dt : `~astropy.units.Quantity`
+        Desired interval between the input map and returned map.
+
+    Returns
+    -------
+   `~sunpy.map`
+        A map with the result of applying solar differential rotation to the
+        input map.
+    """
+    # Check for masked maps
+    if smap.mask is not None:
+        smap_data = np.ma.array(smap.data, mask=smap.mask)
+    else:
+        smap_data = smap.data
+
+    # Apply solar differential rotation as a scikit-image warp
+    out = transform.warp(_to_norm(smap_data), inverse_map=_warp_sun,
+                         map_args={'smap': smap, 'dt': dt})
+
+    # Recover the original intensity range.
+    out = _un_norm(out.T, smap.data)
+
+    # Update the meta information with the new date and time.
+    out_meta = deepcopy(smap.meta)
+    date_keys = ('date-obs', 'date_obs')
+    date_key_flag = False
+    for k in date_keys:
+        if k in out_meta:
+            out_meta[k] = smap.date + timedelta(seconds=dt.to(u.s).value)
+            date_key_flag = True
+    if not date_key_flag:
```

Comment:
One thing I don't understand it's how a map is not going to have either `date_obs` or `date-obs` and have a `.date` attribute. @wafels  - you wrote that... what did you have in mind?